### PR TITLE
feat(Guild#fetch): withCount param

### PR DESCRIPTION
**Pull request by Fyko**
**Please describe the changes this PR makes and why it should be merged:**
This PR adds support for the new **`GET`** `guilds/:id` query parameter `with_counts` per https://github.com/discord/discord-api-docs/pull/1528. This includes two new properties: `approximate_member_count` and `approximate_presence_count` both assigned to `Guild#approximateMemberCount` and `Guild#approximatePresenceCount` respectively. 

I can add support for the `with_counts` query param for `GuildManager#fetch` once #4086 is merged.

This PR will be a draft until the feature is deployed. ~~I am still [yet to test this](https://discordapp.com/channels/613425648685547541/697489244649816084/701861149112664064).~~

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by Fyko** [comment]
Just tested this - works as anticipated. Marking as ready for review. 👍

File: src/structures/Guild.js [hunk 1]
```diff
@@ -305,6 +305,26 @@ class Guild extends Base {
...
     if (typeof data.max_presences !== 'undefined') this.maximumPresences = data.max_presences || 25000;
 
+    /**
+     * The approximate amount of members the guild has
+     * <info>You will need to fetch the guild using {@link Guild#fetch} if you want to receive this parameter</info>
+     * @type {?number}
+     * @name Guild#approximateMemberCount
+     */
+    if (typeof data.approximate_member_count !== 'undefined') {
+      this.approximateMemberCount = data.approximate_member_count || null;
```
Comment (NotSugden) [inline src/structures/Guild.js:315]: wouldn't this be undefined if the guild wasnt fetched? however the typings say that it should be null
or is the approximate_member_count field sent as null in the initial GUILD_CREATE?

File: src/structures/Guild.js [hunk 2]
```diff
@@ -305,6 +305,26 @@ class Guild extends Base {
...
+    }
+
+    /**
+     * The approximate amount of presences the guild has
+     * <info>You will need to fetch the guild using {@link Guild#fetch} if you want to receive this parameter</info>
+     * @type {?number}
+     * @name Guild#approximatePresenceCount
+     */
+    if (typeof data.approximate_presence_count !== 'undefined') {
+      this.approximatePresenceCount = data.approximate_presence_count || null;
```
Comment (NotSugden) [inline src/structures/Guild.js:325]: same as above

File: src/structures/Guild.js [hunk 3]
```diff
@@ -585,12 +605,13 @@ class Guild extends Base {
...
    * Fetches this guild.
+   * @param {boolean} [withCounts=false] Wether or not to fetch the approximate member and presence count
    * @returns {Promise<Guild>}
    */
-  fetch() {
+  fetch(withCounts = false) {
     return this.client.api
       .guilds(this.id)
-      .get()
+      .get({ query: { with_counts: withCounts } })
```
Comment (SpaceEEC) [inline src/structures/Guild.js:614]: To stay consistent to `Client#fetchInvite` this should probably be just `true` and not a parameter.
https://github.com/discordjs/discord.js/blob/88133d0d774e703700fd92ea0a3a63cc78ac4f22/src/client/Client.js#L248-L254

**Pull request merged**